arrow-ord/src/sort.rs

```diff
@@ -303,18 +303,118 @@ fn sort_bytes<T: ByteArrayType>(
     sort_impl(options, &mut valids, &nulls, limit, Ord::cmp).into()
 }
 
+/// Builds a 128-bit composite key for an inline value:
```

Comment:
This comment does a good job explaining what this function does but I think it would be good if we could explain the *why* it is useful a bit more specifically than "fast comparisons". Spcifically what property(s) the returned `u128` has

Something like this 
```suggestion
/// Builds a 128-bit composite key for an inline value for fast sorting
/// 
/// The `u128` returned by this function compares the same comparing
/// the `str` value from an inlined view. 
```

I am not quite sure is that is correct because if it is, I don't understand why it is copying the length bytes as well 🤔